internal/kafka/test/mocks/clusters/clusters.go

```diff
@@ -0,0 +1,64 @@
+package mocks
+
+import (
+	"encoding/json"
+	"time"
+
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/kafka/internal/clusters/types"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/api"
+)
+
+var (
+	testRegion        = "us-west-1"
+	testProvider      = "aws"
+	testCloudProvider = "aws"
+	testMultiAZ       = true
+	testStatus        = api.ClusterProvisioned
+	testClusterID     = "123"
+)
+
+// build a test cluster
+func BuildCluster(modifyFn func(cluster *api.Cluster)) *api.Cluster {
+	clusterSpec, _ := json.Marshal(&types.ClusterSpec{})
+	providerSpec, _ := json.Marshal("{}")
+	cluster := &api.Cluster{
+		Region:        testRegion,
+		CloudProvider: testProvider,
+		MultiAZ:       testMultiAZ,
+		ClusterID:     testClusterID,
+		ExternalID:    testClusterID,
+		Status:        testStatus,
+		ClusterSpec:   clusterSpec,
+		ProviderSpec:  providerSpec,
+	}
+	if modifyFn != nil {
+		modifyFn(cluster)
+	}
+	return cluster
+}
```

Comment:
I believe this would be a good candidate for the `Functional Options Pattern)`:
```go
type ClusterOption func(*api.Cluster)

func WithRegion(region string) ClusterOption {
  return func(cluster *api.Cluster) {
    cluster.Region = region
  }
}

func WithProvider(provider string) ClusterOption {
  return func(cluster *api.Cluster) {
    cluster.Provider = provider
  }
} 

...

func BuildCluster(options ...ClusterOption) *api.Cluster {
  clusterSpec, _ := json.Marshal(&types.ClusterSpec{})
	  providerSpec, _ := json.Marshal("{}")
	  cluster := &api.Cluster{
		  Region:        testRegion,
		  CloudProvider: testProvider,
		  MultiAZ:       testMultiAZ,
		  ClusterID:     testClusterID,
		  ExternalID:    testClusterID,
		  Status:        testStatus,
		  ClusterSpec:   clusterSpec,
		  ProviderSpec:  providerSpec,
	  }
    for _, option := range options {
      option(cluster)
    }
  return cluster
}

```

With this pattern, if we want to create a `Cluster` object using default values, we just have to call:
```go
cluster := BuildCluster()
```

while if we want to customize it:
```go
cluster := BuildCluster(
  WithProvider("testProvider"),
  WithRegion("testRegion"),
)
```

I believe this way the code is much more readable.